File: tests/tester_correspondence/cases/001_empty_report/case.php
#!/usr/bin/env php
<?php

declare(strict_types=1);

$repoRoot = dirname(__DIR__, 4);
require $repoRoot . '/php/tester/vendor/autoload.php';

use IPP\Tester\Model\TestReport;

$report = new TestReport(discoveredTestCases: [], unexecuted: [], results: []);
$json = json_encode($report->toArray(), JSON_PRETTY_PRINT);
if (!is_string($json)) {
    throw new RuntimeException('Failed to serialize JSON.');
}

echo $json . PHP_EOL;

File: tests/tester_correspondence/cases/001_empty_report/case.py
#!/usr/bin/env python3

from __future__ import annotations

import sys
from pathlib import Path

REPO_ROOT = Path(__file__).resolve().parents[4]
sys.path.insert(0, str(REPO_ROOT / "python" / "tester" / "src"))

from models import TestReport  # noqa: E402


report = TestReport(discovered_test_cases=[], unexecuted={}, results={})
print(report.model_dump_json(indent=2))
